lib/web/conn_upgrade.go

```diff
@@ -326,3 +326,15 @@ func (c *websocketALPNServerConn) SetDeadline(t time.Time) error {
 		c.Conn.SetWriteDeadline(t),
 	)
 }
+
+func (c *websocketALPNServerConn) SetWriteDeadline(t time.Time) error {
```

Comment:
We are already using mutex for `SetDeadline` above. Forgot to implement `SetWriteDeadline` and `SetReadDeadline`